### PR TITLE
fix: Attempt to assign property type on null error

### PR DESCRIPTION
**Pull request by ishanvyas22**
Fixes https://github.com/rectorphp/rector/issues/6725

This error is raised when there's discrepancies between params and phpDocParams. That means, for example, if method accepts 2 params and phpDoc block is only added for any one of param, in that case this issue is raised.

The reason for the issue is depending on [phpDoc $key variable](https://github.com/rectorphp/rector-src/blob/main/rules/CodeQuality/Rector/ClassMethod/NarrowUnionTypeDocRector.php#L126). I tried to check it with param name but it added a duplicated `@param` tag in docBlock. So as a temporary fix I added `isset` check for now, so there's no notice/error is raised.

### CODE REVIEW

File: rules-tests/CodeQuality/Rector/ClassMethod/NarrowUnionTypeDocRector/Fixture/bug_6725.php.inc [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?php
```
Comment (samsonasik) [inline rules-tests/CodeQuality/Rector/ClassMethod/NarrowUnionTypeDocRector/Fixture/bug_6725.php.inc]: Please rename fixture file, eg: `overlapped_key_param.php.inc`

Comment (ishanvyas22) [reply rules-tests/CodeQuality/Rector/ClassMethod/NarrowUnionTypeDocRector/Fixture/bug_6725.php.inc]: Renamed it.

File: rules-tests/CodeQuality/Rector/ClassMethod/NarrowUnionTypeDocRector/Fixture/overlapped_key_param.php.inc [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class Factory
+{
+    /**
+     * @param string|int|float|bool $value
+     */
+    public function create(string $name, $value): int
+    {
+
+    }
+}
```
Comment (samsonasik) [inline rules-tests/CodeQuality/Rector/ClassMethod/NarrowUnionTypeDocRector/Fixture/overlapped_key_param.php.inc]: Before and after seems equal, you can remove the expected output:

```suggestion
```

File: rules-tests/CodeQuality/Rector/ClassMethod/NarrowUnionTypeDocRector/Fixture/overlapped_key_param.php.inc [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?php
```
Comment (samsonasik) [inline rules-tests/CodeQuality/Rector/ClassMethod/NarrowUnionTypeDocRector/Fixture/overlapped_key_param.php.inc]: one more change, could you add `skip_` prefix to the fixture file, thank you

Comment (ishanvyas22) [reply rules-tests/CodeQuality/Rector/ClassMethod/NarrowUnionTypeDocRector/Fixture/overlapped_key_param.php.inc]: Is it to skip for windows OS?

**Comment by samsonasik** [comment]
Thank you @ishanvyas22

File: rules-tests/CodeQuality/Rector/ClassMethod/NarrowUnionTypeDocRector/Fixture/overlapped_key_param.php.inc (hunk 3 above)
Comment (samsonasik) [reply rules-tests/CodeQuality/Rector/ClassMethod/NarrowUnionTypeDocRector/Fixture/overlapped_key_param.php.inc]: that's just for naming convention to make consistent for skipped fixture.

Comment (ishanvyas22) [reply rules-tests/CodeQuality/Rector/ClassMethod/NarrowUnionTypeDocRector/Fixture/overlapped_key_param.php.inc]: Got it

**Comment by ishanvyas22** [comment]
Thanks. Would it be possible to add "hacktoberfest-accepted" label to this PR?

**Pull request merged**